### ISSUE TITLE
PYTHON backend does not handle overflow half

### ISSUE THREAD

**Issue by chenyuxyz** [tinygrad/tinygrad#5018]
more problematic than just PYTHON because `Tensor(list)` is handled by PYTHON now (instead of numpy), for example this fails
```
python -c "from tinygrad import Tensor, dtypes; print(Tensor([80000.0], dtype=dtypes.float16).numpy())"
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/Users/chenyu/code/tinygrad/tinygrad/tensor.py", line 121, in __init__
    else: data = _frompy(data, dtype)
                 ^^^^^^^^^^^^^^^^^^^^
  File "/Users/chenyu/code/tinygrad/tinygrad/tensor.py", line 54, in _frompy
    data = struct.pack(f"@{ret.size}{dtype.fmt}", *fully_flatten(x))
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
OverflowError: float too large to pack with e format
```

could not find a built-in method to convert this, might need to manually manipulates the bits (would need this for bfloat16 anyway)

**Comment by chenyuxyz** [comment]
how cpython packs float https://github.com/python/cpython/blob/ac37a806018cc40fafebcd0fa90250c3e0261e0c/Objects/floatobject.c#L2048

**Comment by chenyuxyz** [comment]
fixed by #5022. catch the struct.pack OverflowError for now
